controllers/events.js

```diff
@@ -16,24 +25,33 @@ const eventTypes = Object.freeze({
   logged : "TIME_TRACKED"
 });
 
+const validateTrackingData = (e) => {
+  if(e.trackingType == "UNTRACKED" && !util.string.isNullOrEmpty(e.trackingData)) return "";
+  if(e.trackingType == "TRACKED" && !(e.trackingData == "0" || e.trackingData == "1")) return ec.events.INVALID_TRACKING_DATA;
+  if(e.trackingType == "TIME_TRACKED" && parseFloat(e.trackingData) >= 0 && parseFloat(e.trackingData) <= 1) return ec.events.INVALID_TRACKING_DATA; 
+  return "";
+}; 
+
 const validateEvent = (e) => {
   var error = "";
-  if( !e.name || 
-    !e.date || 
-    !e.startTime || 
-    !e.endTime || 
-    !Object.values(eventTypes).includes(e.trackingType)) {
-    error = ec.events.INCOMPLETE; 
-  } 
-  else if(!util.date.isValid(e.date)) error =  ec.events.INVALID_DATE_FORMAT; 
-  else if(!util.time.isValid(e.startTime)) error = ec.events.INVALID_TIME_FORMAT;
-  else if(!util.time.isValid(e.endTime)) error = ec.events.INVALID_TIME_FORMAT;
-  else if(!util.time.isBeforeTime(e.startTime,e.endTime)) error = ec.events.INVALID_TIME_VALUES;
+  const valid = validator.validate(e, eventSchema).valid;
+  if(!valid) error = ec.events.INCOMPLETE;
+  else if(!util.date.isValid(e.date)) error =  ec.events.INVALID_DATE_FORMAT;
+  else if(validatebaseEvent(e.baseEvent) != "") error = validatebaseEvent(e.baseEvent);
+  else if(validateTrackingData(e) != "") error = validateTrackingData(e);
+  return error;
+};
+
+export const validatebaseEvent = (baseEvent) => {
+  var error = "";
+  if(!util.time.isBeforeTime(baseEvent.startTime,baseEvent.endTime)) error = ec.events.INVALID_TIME_VALUES; 
+  else if(!util.time.isValid(baseEvent.startTime) || !util.time.isValid(baseEvent.endTime))  error = ec.events.INVALID_TIME_FORMAT;
```

Comment:
Check `baseEvent.startTime` and `baseEvent.endTime` are valid before operating on them (`isBeforeTime(t1, t2)`)